sdl2promela/stop_condition/translator.py

```diff
@@ -246,9 +299,185 @@ def _generate(expr: model.Selector):
     return result
 
 
-@dispatch(model.CallExpression)
-def _generate(expr: model.CallExpression):
-    raise TranslateException("Not implemented.")
+def _resolve_type(allTypes, t):
+    while t.kind == "ReferenceType":
+        if t.ReferencedTypeName not in allTypes:
+            raise ("Cannot resolve type {}".format(t.ReferencedTypeNam))
+        t = allTypes[t.ReferencedTypeName].type
+
+    return t
+
+
+@dispatch(GenerateContext, model.Selector)
+def _find_type(context: GenerateContext, selector: model.Selector):
+    index = 0
+    if isinstance(selector.elements[index], model.CallExpression):
+        finalType, allTypes = _find_type(context, selector.elements[index])
+        index = index + 1
```

Comment:
3 consecutive index increments do not seem conditional.... Would it be possible to use descriptive constants?

Reply:
Refactored a little bit. I added comments, there are two special cases how to begin finding type of selector.

Reply:
Refactored again